File: ios/Models/SegmentationOutput.swift
import Foundation

struct SegmentationOutput {
  let rgba: Data
  let width: Int
  let height: Int
  let bounds: Rect
  let sourceWidth: Int
  let sourceHeight: Int
  let foregroundCoverage: Double
  let centroid: NormalizedPoint
  let instanceCount: Double
  let pixelBounds: PixelRect
  let trimOrigin: NormalizedPoint
  let mask: Data
  let hasMask: Bool
}

File: android/src/main/java/com/margelo/nitro/nitrovisionkit/models/SegmentationOutput.kt
package com.margelo.nitro.nitrovisionkit

internal class SegmentationOutput(
  val pixels: ByteArray,
  val width: Int,
  val height: Int,
  val bounds: VisionRect,
  val sourceWidth: Int,
  val sourceHeight: Int,
  val foregroundCoverage: Double,
  val centroid: NormalizedPoint,
  val instanceCount: Double,
  val pixelBounds: PixelRect,
  val trimOrigin: NormalizedPoint,
  val mask: ByteArray,
  val hasMask: Boolean,
)
